From 5766ab6ed3565f51947bc5c3b1669c307d64984d Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 7 Jul 2022 15:44:20 +0200
Subject: [PATCH 62/95] arm64: dts: st: add scmi performance support on
 stm32mp25

Update the device-tree to add support for scmi performance
on cpu0 and cpu1. This leads to enable cpu-freq on the CPUs.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 477b38716186f5e1fd9149a4863b9e374f0a8091 arm64: dts: st: add scmi performance support on stm32mp25]
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Change-Id: I853c03c86900c5959e93f42534a5b8bd72cf6ea0
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 7 +++++++
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 37b69bfd25f43..6a08816acd9e3 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -25,6 +25,8 @@ cpu0: cpu@0 {
 			enable-method = "psci";
 			power-domains = <&CPU_PD0>;
 			power-domain-names = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
@@ -75,6 +77,11 @@ scmi {
 			#size-cells = <0>;
 			linaro,optee-channel-id = <0>;
 
+			scmi_perf: protocol@13 {
+				reg = <0x13>;
+				#clock-cells = <1>;
+			};
+
 			scmi_clk: protocol@14 {
 				reg = <0x14>;
 				#clock-cells = <1>;
diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index eeceb086252bf..6d5c6d36ad616 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -14,6 +14,8 @@ cpu1: cpu@1 {
 			enable-method = "psci";
 			power-domains = <&CPU_PD1>;
 			power-domain-names = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
-- 
2.51.0

